sky/utils/vpn_utils.py

```diff
@@ -0,0 +1,285 @@
+
+    def __init__(
+        self,
+        auth_key: str,
+        api_key: Optional[str] = None,
+        network_name: Optional[str] = None,
+        enable_api: bool = False,
+    ) -> None:
+        self._auth_key = auth_key
+        self._api_key = api_key
+        self._network_name = network_name
+        self._enable_api = enable_api
+        if not enable_api:
+            logger.warning('TAILSCALE_API_KEY or TAILSCALE_NETWORK_NAME is'
+                           ' not set. Tailscale API will be disabled. You'
+                           ' may need to remove hosts manually.')
+        elif not api_key or not network_name:
+            raise ValueError('Both TAILSCALE_API_KEY and TAILSCALE_NETWORK_NAME'
+                             ' must be set to enable the Tailscale API.')
+
+    @staticmethod
+    def from_env_vars() -> 'TailscaleVPN':
+        # Parse Tailscale auth key from environment variable.
+        # This is required for all Tailscale operations.
+        auth_key = os.environ.get('TAILSCALE_AUTH_KEY')
+        if not auth_key:
+            raise ValueError('TAILSCALE_AUTH_KEY is not set.')
+
+        # Parse Tailscale API key and network name
+        api_key = os.environ.get('TAILSCALE_API_KEY')
+        network_name = os.environ.get('TAILSCALE_NETWORK_NAME')
+        enable_api = True
+        if not api_key or not network_name:
+            enable_api = False
+
+        return TailscaleVPN(auth_key, api_key, network_name, enable_api)
+
+    def get_setup_command(self) -> str:
+        return self._SETUP_COMMAND.format(tailscale_auth_key=self._auth_key)
+
+    def get_setup_env_vars(self) -> Dict[str, str]:
+        env_config = {
+            'TAILSCALE_AUTH_KEY': self._auth_key,
+        }
+        if self._enable_api:
+            assert self._api_key and self._network_name
+            env_config['TAILSCALE_API_KEY'] = self._api_key
+            env_config['TAILSCALE_NETWORK_NAME'] = self._network_name
+        return env_config
+
+    def _get_auth_headers(self) -> Dict[str, str]:
+        """Get the authentication headers for the Tailscale API."""
+        return {
+            'Authorization': f'Bearer {self._api_key}',
+        }
+
+    def _get_device_id_from_hostname(self, hostname: str) -> Optional[str]:
+        """Get the node ID from the hostname."""
+        url_to_query = ('https://api.tailscale.com/api/v2/tailnet/'
+                        f'{self._network_name}/devices')
+        resp = requests.get(url_to_query, headers=self._get_auth_headers())
+        all_devices_in_network = resp.json().get('devices', [])
+        for device_info in all_devices_in_network:
+            if device_info.get('hostname') == hostname:
+                return device_info.get('id')
+        return None
+
+    def get_private_ip(self, hostname: str) -> str:
+        """Get the private IP address from the hostname."""
+        query_cmd = f'tailscale ip -4 {hostname}'
```

Comment:
please investigate if this can be done using REST API. This require user to locally install the VPN. If this can only be done using the tailscale CLI, then we need to check if it is installed in local laptop and raise an error if not.